app/models/concerns/blacklight/solr/document/marc.rb

```diff
@@ -17,72 +19,33 @@ def to_marc
-          unless format.nil?
-            format = format.is_a?(Array) ? format[0].downcase.strip : format.downcase.strip
-            genre = format_to_openurl_genre(format)
-          end
-          export_text = ''
-          if format == 'book'
-            export_text << 'ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Abook&amp;rfr_id=info%3Asid%2Fpulsearch.princeton.edu%3Agenerator&amp;rft.genre=book&amp;'
-            export_text << "rft.btitle=#{(title.nil? || title['a'].nil?) ? '' : CGI.escape(title['a'])}+#{(title.nil? || title['b'].nil?) ? '' : CGI.escape(title['b'])}&amp;"
-            export_text << "rft.title=#{(title.nil? || title['a'].nil?) ? '' : CGI.escape(title['a'])}+#{(title.nil? || title['b'].nil?) ? '' : CGI.escape(title['b'])}&amp;"
-            export_text << "rft.au=#{(author.nil? || author['a'].nil?) ? '' : CGI.escape(author['a'])}&amp;"
-            export_text << "rft.aucorp=#{CGI.escape(corp_author['a']) if corp_author['a']}+#{CGI.escape(corp_author['b']) if corp_author['b']}&amp;" unless corp_author.blank?
-            export_text << "rft.date=#{(publisher_info.nil? || publisher_info['c'].nil?) ? '' : CGI.escape(publisher_info['c'])}&amp;"
-            export_text << "rft.place=#{(publisher_info.nil? || publisher_info['a'].nil?) ? '' : CGI.escape(publisher_info['a'])}&amp;"
-            export_text << "rft.pub=#{(publisher_info.nil? || publisher_info['b'].nil?) ? '' : CGI.escape(publisher_info['b'])}&amp;"
-            export_text << "rft.edition=#{(edition.nil? || edition['a'].nil?) ? '' : CGI.escape(edition['a'])}&amp;"
-            export_text << "rft.isbn=#{(isbn.nil? || isbn['a'].nil?) ? '' : isbn['a']}"
-            export_text << '&amp;rft.genre=book'
-          elsif format =~ /journal/i # checking using include because institutions may use formats like Journal or Journal/Magazine
-            export_text << 'ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rfr_id=info%3Asid%2Fblacklight.rubyforge.org%3Agenerator&amp;rft.genre=article&amp;'
-            export_text << "rft.title=#{(title.nil? || title['a'].nil?) ? '' : CGI.escape(title['a'])}+#{(title.nil? || title['b'].nil?) ? '' : CGI.escape(title['b'])}&amp;"
-            export_text << "rft.atitle=#{(title.nil? || title['a'].nil?) ? '' : CGI.escape(title['a'])}+#{(title.nil? || title['b'].nil?) ? '' : CGI.escape(title['b'])}&amp;"
-            export_text << "rft.aucorp=#{CGI.escape(corp_author['a']) if corp_author['a']}+#{CGI.escape(corp_author['b']) if corp_author['b']}&amp;" unless corp_author.blank?
-            export_text << "rft.date=#{(publisher_info.nil? || publisher_info['c'].nil?) ? '' : CGI.escape(publisher_info['c'])}&amp;"
-            export_text << "rft.issn=#{(issn.nil? || issn['a'].nil?) ? '' : issn['a']}"
-            export_text << '&amp;rft.genre=serial'
+        # returns true if doc originated from voyager
+        def voyager_record?
```

Comment:
This is great, now we don't have to hit bibdata every time we call this method. I am wondering what will happen though when the id is a voyager id and the record has been deleted.